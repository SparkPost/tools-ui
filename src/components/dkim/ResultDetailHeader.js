import React, { Component } from 'react';
import { CopyPopover } from 'components/popover/Popover';
import { ActionLink } from 'components/button/Button';

import './ResultDetailHeader.scss';

class ResultDetailHeader extends Component {
  constructor(props) {
    super(props);
  }

  renderDetails() {
    const { rows } = this.props;
    return (
      rows.map((row, key) => (
        <div className='dkimResultDetailHeader' key={key}>
          {row.map((value, key) => {
            if (key === 0) {
              return <span className='dkimResultDetailHeader__label' key={key}>{value}</span>;
            }
            return <span className='dkimResultDetailHeader__value' key={key}>{value}</span>;
          })}
        </div>
      ))
    );
  }

  render() {
    const { loggedIn } = this.props;

    return (
      <div className='panel panel--accent'>
        <div className='panel__body'>
          <div className='float--right'>
            {!loggedIn && <ActionLink to='/'>Save Results</ActionLink>}
            <CopyPopover>
              <ActionLink title='Share'>Share</ActionLink>
            </CopyPopover>
          </div>
          {this.renderDetails()}
        </div>
      </div>
    );
  }
}

ResultDetailHeader.defaultProps = {
  loggedIn: false
};

ResultDetailHeader.propTypes = {
  loggedIn: React.PropTypes.bool.isRequired
};

export default ResultDetailHeader;
