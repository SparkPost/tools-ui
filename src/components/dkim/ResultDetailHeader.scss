@import "~@sparkpost/styles/src/assets/scss/shared/config";
@import "~@sparkpost/styles/src/assets/scss/shared/functions";

.dkimResultDetailHeader {
  &__value, &__label {
    display: inline-block;
    line-height: rem(21);
    font-weight: 500;
  }

  &__value {
    color: grayscale(2);
    font-size: rem(14);
  }

  &__label {
    width: rem(42);
    margin-right: rem(18);

    text-align: right;
    font-size: rem(12);
    color: grayscale(5);
    text-transform: lowercase;
  }
}
