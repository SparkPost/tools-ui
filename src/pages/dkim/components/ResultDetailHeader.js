import React from 'react';
import { CopyPopover } from 'components/popover/Popover';
import { ActionLink, SaveResultsLink } from 'components/button/Button';
import Icon from 'components/Icon';

import './ResultDetailHeader.scss';

const HeaderRow = ({ rows }) => (
  <span>
    {rows.map((row, key) => (
      <div className='dkimResultDetailHeader' key={key}>
        {row.map((value, key) => {
          const type = key === 0 ? 'label' : 'value';
          return <span key={key} className={`dkimResultDetailHeader__${type}`}>{value}</span>;
        })}
      </div>
    ))}
  </span>
);

const HeaderStatus = ({ status }) => {
  const icon = status === 'Passed' ? 'check-circle' : 'exclamation-circle';
  const type = status === 'Passed' ? 'is-valid' : 'has-error';
  return (
    <div className='panel__body'>
      <span className='dkimResultDetailHeader__label'>status</span>
      <span className={`dkimResultDetailHeader__value ${type}`}><Icon name={icon}/> {status}</span>
    </div>
  );
};

const ResultDetailHeader = ({ rows, loggedIn, status, error }) => (
  <div className='panel panel--accent'>
    <div className='panel__body'>
      <div className='float--right'>
        {!loggedIn && <SaveResultsLink/>}
        <CopyPopover>
          <ActionLink title='Share' track={true}>Share</ActionLink>
        </CopyPopover>
      </div>
      <HeaderRow rows={rows} />
    </div>
    <HeaderStatus status={status} error={error} />
  </div>
);

export default ResultDetailHeader;
