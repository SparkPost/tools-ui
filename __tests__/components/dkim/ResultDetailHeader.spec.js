import React from 'react';
import renderer from 'react-test-renderer';

import ResultDetailHeader from 'components/dkim/ResultDetailHeader';

describe('ResultDetailHeader component', () => {

  const testRows = [
    [1, 2, 3],
    ['a', 'b', 'c'],
    ['cool', 'story', 'bro']
  ];

  test('should render with rows', () => {
    expect(renderer.create(<ResultDetailHeader rows={testRows} />)).toMatchSnapshot();
  });

  test('should render when logged in', () => {
    expect(renderer.create(<ResultDetailHeader rows={testRows} loggedIn={true} />)).toMatchSnapshot();
  });

  test('should render when not logged in', () => {
    expect(renderer.create(<ResultDetailHeader rows={testRows} loggedIn={false} />)).toMatchSnapshot();
  });

});
