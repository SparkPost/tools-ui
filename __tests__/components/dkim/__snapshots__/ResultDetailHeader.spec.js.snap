exports[`ResultDetailHeader component should render when logged in 1`] = `
<div
  className="panel panel--accent">
  <div
    className="panel__body">
    <div
      className="float--right">
      <span
        className="popover__group"
        onClick={[Function]}>
        <a
          className="actionLink"
          onClick={[Function]}
          style={Object {}}
          title="Share">
          Share
        </a>
        <div
          className="popover copyPopover popover--l popover--rightAligned popover--bottom">
          <h6
            className="text--left copyPopover__label">
            <i
              className="fa fa-check-circle " />
             Copied to Clipboard!
          </h6>
          <p
            className="copyPopover__copiedString">
            http://localhost/
          </p>
          <input
            className="copyPopover__hidden"
            readOnly={true}
            value="http://localhost/" />
        </div>
      </span>
    </div>
    <span>
      <div
        className="dkimResultDetailHeader">
        <span
          className="dkimResultDetailHeader__label">
          1
        </span>
        <span
          className="dkimResultDetailHeader__value">
          2
        </span>
        <span
          className="dkimResultDetailHeader__value">
          3
        </span>
      </div>
      <div
        className="dkimResultDetailHeader">
        <span
          className="dkimResultDetailHeader__label">
          a
        </span>
        <span
          className="dkimResultDetailHeader__value">
          b
        </span>
        <span
          className="dkimResultDetailHeader__value">
          c
        </span>
      </div>
      <div
        className="dkimResultDetailHeader">
        <span
          className="dkimResultDetailHeader__label">
          cool
        </span>
        <span
          className="dkimResultDetailHeader__value">
          story
        </span>
        <span
          className="dkimResultDetailHeader__value">
          bro
        </span>
      </div>
    </span>
  </div>
  <div
    className="panel__body">
    <div
      className="float--right">
      <a
        className="actionLink"
        href="https://support.sparkpost.com/tools-help"
        target="_blank"
        title="">
        How do I fix errors?
      </a>
    </div>
    <span
      className="dkimResultDetailHeader__label">
      status
    </span>
    <span
      className="dkimResultDetailHeader__value has-error">
      <i
        className="fa fa-exclamation-circle " />
       
    </span>
  </div>
</div>
`;

exports[`ResultDetailHeader component should render when not logged in 1`] = `
<div
  className="panel panel--accent">
  <div
    className="panel__body">
    <div
      className="float--right">
      <span
        className="popover__group popover__hoverTrigger">
        <a
          className="actionLink"
          href="https://TEST.sparkpost.com/sign-up?return=http%3A%2F%2Flocalhost%2F&src=SP-Tools&sfdcid=701600000019BWh"
          target="_self"
          title="Save Results">
          Save Results
        </a>
        <span
          className="popover popover--top popover--m">
          Create a free SparkPost account or login into your account to save results
        </span>
      </span>
      <span
        className="popover__group"
        onClick={[Function]}>
        <a
          className="actionLink"
          onClick={[Function]}
          style={Object {}}
          title="Share">
          Share
        </a>
        <div
          className="popover copyPopover popover--l popover--rightAligned popover--bottom">
          <h6
            className="text--left copyPopover__label">
            <i
              className="fa fa-check-circle " />
             Copied to Clipboard!
          </h6>
          <p
            className="copyPopover__copiedString">
            http://localhost/
          </p>
          <input
            className="copyPopover__hidden"
            readOnly={true}
            value="http://localhost/" />
        </div>
      </span>
    </div>
    <span>
      <div
        className="dkimResultDetailHeader">
        <span
          className="dkimResultDetailHeader__label">
          1
        </span>
        <span
          className="dkimResultDetailHeader__value">
          2
        </span>
        <span
          className="dkimResultDetailHeader__value">
          3
        </span>
      </div>
      <div
        className="dkimResultDetailHeader">
        <span
          className="dkimResultDetailHeader__label">
          a
        </span>
        <span
          className="dkimResultDetailHeader__value">
          b
        </span>
        <span
          className="dkimResultDetailHeader__value">
          c
        </span>
      </div>
      <div
        className="dkimResultDetailHeader">
        <span
          className="dkimResultDetailHeader__label">
          cool
        </span>
        <span
          className="dkimResultDetailHeader__value">
          story
        </span>
        <span
          className="dkimResultDetailHeader__value">
          bro
        </span>
      </div>
    </span>
  </div>
  <div
    className="panel__body">
    <div
      className="float--right">
      <a
        className="actionLink"
        href="https://support.sparkpost.com/tools-help"
        target="_blank"
        title="">
        How do I fix errors?
      </a>
    </div>
    <span
      className="dkimResultDetailHeader__label">
      status
    </span>
    <span
      className="dkimResultDetailHeader__value has-error">
      <i
        className="fa fa-exclamation-circle " />
       
    </span>
  </div>
</div>
`;

exports[`ResultDetailHeader component should render with rows 1`] = `
<div
  className="panel panel--accent">
  <div
    className="panel__body">
    <div
      className="float--right">
      <span
        className="popover__group popover__hoverTrigger">
        <a
          className="actionLink"
          href="https://TEST.sparkpost.com/sign-up?return=http%3A%2F%2Flocalhost%2F&src=SP-Tools&sfdcid=701600000019BWh"
          target="_self"
          title="Save Results">
          Save Results
        </a>
        <span
          className="popover popover--top popover--m">
          Create a free SparkPost account or login into your account to save results
        </span>
      </span>
      <span
        className="popover__group"
        onClick={[Function]}>
        <a
          className="actionLink"
          onClick={[Function]}
          style={Object {}}
          title="Share">
          Share
        </a>
        <div
          className="popover copyPopover popover--l popover--rightAligned popover--bottom">
          <h6
            className="text--left copyPopover__label">
            <i
              className="fa fa-check-circle " />
             Copied to Clipboard!
          </h6>
          <p
            className="copyPopover__copiedString">
            http://localhost/
          </p>
          <input
            className="copyPopover__hidden"
            readOnly={true}
            value="http://localhost/" />
        </div>
      </span>
    </div>
    <span>
      <div
        className="dkimResultDetailHeader">
        <span
          className="dkimResultDetailHeader__label">
          1
        </span>
        <span
          className="dkimResultDetailHeader__value">
          2
        </span>
        <span
          className="dkimResultDetailHeader__value">
          3
        </span>
      </div>
      <div
        className="dkimResultDetailHeader">
        <span
          className="dkimResultDetailHeader__label">
          a
        </span>
        <span
          className="dkimResultDetailHeader__value">
          b
        </span>
        <span
          className="dkimResultDetailHeader__value">
          c
        </span>
      </div>
      <div
        className="dkimResultDetailHeader">
        <span
          className="dkimResultDetailHeader__label">
          cool
        </span>
        <span
          className="dkimResultDetailHeader__value">
          story
        </span>
        <span
          className="dkimResultDetailHeader__value">
          bro
        </span>
      </div>
    </span>
  </div>
  <div
    className="panel__body">
    <div
      className="float--right">
      <a
        className="actionLink"
        href="https://support.sparkpost.com/tools-help"
        target="_blank"
        title="">
        How do I fix errors?
      </a>
    </div>
    <span
      className="dkimResultDetailHeader__label">
      status
    </span>
    <span
      className="dkimResultDetailHeader__value has-error">
      <i
        className="fa fa-exclamation-circle " />
       
    </span>
  </div>
</div>
`;
